src/logplex_tlssyslog_drain.erl

```diff
@@ -404,6 +405,23 @@ do_reconnect(State = #state{sock = undefined,
                     %% first failure.
                     ok;
                 _ ->
+                    % inject log message
+                    Msg =  iolist_to_binary(
+                               logplex_syslog_utils:to_msg(
+                                 logplex_syslog_utils:fmt(local7,
+                                                          warning,
+                                                          now,
+                                                          "heroku", 
+                                                          "logplex",
+                                                          "Error L14 (certificate validation): "
+                                                          "error=\"~s" uri=\"~s\"",
+                                                          [error_to_human(Reason), drain_uri(State)]),
+                                 State#state.drain_tok)),
+                    logplex_message:process_msg(Msg,
+                                                State#state.channel_id,
+                                                <<"heroku">>,
+                                                <<"heroku">>,
+                                                logplex_message:shard_info()),
```

Comment:
I can simplify this code and not export `shard_info/0`. There are however two use cases

1. messages from http input
2. messages from syslog input

The former assumes the same shard info for all logs in the frame. The latter cannot make that assumption and must process line by line and refetch shard info for each token.